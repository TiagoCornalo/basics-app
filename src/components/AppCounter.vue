<template>
  <div>
    <p data-testid="counter">{{ count }}</p>
    <button @click="count++">Increment</button>
    <button @click="count--">Decrement</button>
    <button @click.alt="count += 10">Increment 10</button>
  </div>
  <div>
    <h2>
      {{ customTitle }}
    </h2>
    <p> {{ count }} <sup>2</sup> = {{ squareValue }}</p>
    <p>{{ count }}</p>
  </div>
</template>

<script>
export default {
  props: {
    title: String,
    start: {
      type: Number,
      default: 150
    }
  },
  name: 'AppCounter',

  data() {
    return {
      count: this.start
    }
  },

  methods: {
    getSquareValue() {
      return this.count * this.count;
    }
  },

  computed: {
    squareValue() {
      return this.count * this.count;
    },
    customTitle() {
      return this.title || 'Counter';
    }
  }
}
</script>

<style></style>