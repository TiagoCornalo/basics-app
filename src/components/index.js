export { default as AppCounter } from './AppCounter.vue';
export { default as AppIndecision } from './AppIndecision.vue';
