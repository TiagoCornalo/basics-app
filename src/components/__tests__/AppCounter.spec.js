import { describe, it, expect, beforeEach } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import { AppCounter } from '@/components'

describe('AppCounter', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(AppCounter)
  })

  it('should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('h2 title should be "Counter" by default', () => {
    expect(wrapper.find('h2').text()).toBe('Counter')
  })

  it("should display '150' in the paragraph", () => {
    const counterValue = wrapper.find('[data-testid="counter"]')

    expect(counterValue.text()).toBe('150')
  })

  it('should increment counter by 1 when button is clicked', async () => {
    const buttonIncrease = wrapper.find('button')
    await buttonIncrease.trigger('click')

    const counterValue = wrapper.find('[data-testid="counter"]')

    expect(counterValue.text()).toBe('151')
  })

  it('should decrement counter by 1 when button is clicked', async () => {
    const buttonDecrease = wrapper.findAll('button')[1]
    await buttonDecrease.trigger('click')

    const counterValue = wrapper.find('[data-testid="counter"]')

    expect(counterValue.text()).toBe('149')
  })

  it('should increment and decrement the counter', async () => {
    const [buttonIncrease, buttonDecrease] = wrapper.findAll('button')

    await buttonIncrease.trigger('click')

    let counterValue = wrapper.find('[data-testid="counter"]')

    expect(counterValue.text()).toBe('151')

    await buttonDecrease.trigger('click')

    counterValue = wrapper.find('[data-testid="counter"]')

    expect(counterValue.text()).toBe('150')
  })

  it('should have a default value of 150', () => {
    const { start } = wrapper.vm.$options.props
    expect(start.default).toBe(150)
  })

  it('should render the title prop', () => {
    const title = 'New Counter'
    const wrapper = shallowMount(AppCounter, {
      props: {
        title
      }
    })

    expect(wrapper.find('h2').text()).toBe(title)
  })
})
