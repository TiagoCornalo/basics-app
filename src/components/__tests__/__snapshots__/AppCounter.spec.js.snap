// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppCounter > should match snapshot 1`] = `
"<div>
  <p data-testid="counter">150</p><button>Increment</button><button>Decrement</button><button>Increment 10</button>
</div>
<div>
  <h2>Counter</h2>
  <p>150 <sup>2</sup> = 22500</p>
  <p>150</p>
</div>"
`;
